plugin/documents.py

```diff
@@ -482,6 +482,10 @@ def on_hover(self, point: int, hover_zone: int) -> None:
                     on_navigate=lambda href: self._on_navigate(href, point))
 
     def on_text_command(self, command_name: str, args: Optional[dict]) -> Optional[Tuple[str, dict]]:
+        if command_name == 'paste' and userprefs().format_on_paste:
+            session = self.session_async("documentRangeFormattingProvider")
+            if session:
+                return ("lsp_paste_and_format", {})
         if command_name == "auto_complete":
```

Comment:
This can be `elif` instead of `if` now (or alternatively reorder the conditions and use `elif` in the new code).

Reply:
I no longer use the on_text_command, instead I went ahead and implemented your second [comment](https://github.com/sublimelsp/LSP/pull/2311#discussion_r1297531796) to use on_post_text_command